components/asset.rst

```diff
@@ -0,0 +1,251 @@
+methods::
+
+    namespace Symfony\Component\Asset;
+
+    interface PackageInterface
+    {
+        /**
+         * Returns the asset version for an asset.
+         */
+        public function getVersion($path);
+
+        /**
+         * Returns an absolute or root-relative public path.
+         */
+        public function getUrl($path);
+    }
+
+Versioned Assets
+~~~~~~~~~~~~~~~~
+
+One of the main features of the Asset component is to manage the versioning of
+the application's assets. Asset versions are commonly used to control how these
+assets are cached.
+
+Instead of relying on a simple version mechanism, the Asset component allows to
+define advanced versioning strategies via PHP classes. The two built-in strategies
+provided by the component are ``EmptyVersionStrategy``, which doesn't add any
+version to the asset, and ``StaticVersionStrategy``, which allows to set the
+version with a format string.
+
+In this example, the ``StaticVersionStrategy`` is used to append the ``v1``
+suffix to any asset path::
+
+    use Symfony\Component\Asset\Package;
+    use Symfony\Component\Asset\VersionStrategy\StaticVersionStrategy;
+
+    $package = new Package(new StaticVersionStrategy('v1'));
+
+    echo $package->getUrl('/image.png');
+    // result: /image.png?v1
+
+In case you want to modify the version format, pass a sprintf-compatible format
+string as the second argument of the ``StaticVersionStrategy`` constructor::
+
+    // put the 'version' word before the version value
+    $package = new Package(new StaticVersionStrategy('v1', '%s?version=%s'));
+
+    echo $package->getUrl('/image.png');
+    // result: /image.png?version=v1
+
+    // put the asset version before its path
+    $package = new Package(new StaticVersionStrategy('v1', '%2$s/%1$s'));
+
+    echo $package->getUrl('/image.png');
+    // result: /v1/image.png
+
+Custom Version Strategies
+.........................
+
+Use the ``VersionStrategyInterface`` to define your own version strategy. For
+example, you could define a versioning where the current date is appended to
+bust the cache every day::
+
+    use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+
+    class DateVersionStrategy implements VersionStrategyInterface
+    {
+        private $version;
+
+        public function __construct()
+        {
+            $this->version = date('Ymd');
+        }
+
+        public function getVersion($path)
+        {
+            return $this->version;
+        }
+
+        public function applyVersion($path)
+        {
+            return sprintf('%s?v=%s', $path, $this->getVersion($path));
+        }
+    }
+
+Grouped Assets
+~~~~~~~~~~~~~~
+
+It's common for applications to store their assets in a common path. If that's
+your case, replace the default ``Package`` class by ``PathPackage`` to avoid
+repeating the same path time and again::
+
+    use Symfony\Component\Asset\PathPackage;
+
```

Comment:
missing some `// ...` comment (in other examples too) to indicate that not all use statements or boilerplate code is shown